### PR TITLE
Issue168: fix es-node crash when --p2p.disable set

### PR DESCRIPTION
**Pull request by ping-ke**
related issue: https://github.com/ethstorage/es-node/issues/168

How to test:
1. Run p2p sync_test.go
2. Run es-node with `--p2p.disable` param, and es-node can start and close successfully.

### CODE REVIEW

**Review by syntrust — COMMENTED** [review]
I wonder how this change will affect miner to actually start mining? Will sync done event still emitted?

File: ethstorage/storage_manager.go [hunk 1]
```diff
@@ -153,6 +153,10 @@ func (s *StorageManager) Reset(newL1 int64) error {
 	return nil
 }
 
+func (s *StorageManager) GetLastKVIdxByBlockNumber(blockNumber int64) (uint64, error) {
```
Comment (qzhodl) [inline ethstorage/storage_manager.go:156]: what do we need to do this change？

**Comment by qzhodl** [comment]
> I wonder how this change will affect miner to actually start mining? Will sync done event still emitted?

Maybe the miner module can have another flag to indicate start at launch without waiting for the signal?

File: ethstorage/storage_manager.go (hunk 1 above)
Comment (syntrust) [reply ethstorage/storage_manager.go:156]: I think it's ok as long as the behavior is documented

**Review by syntrust — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ The full list of options that you can use to configure an es-node are as follows
 |`--p2p.advertise.udp`|The UDP port to advertise in Discv5 as fallback if not determined by Discv5, put into the ENR of the node. Set to p2p.listen.udp value if 0.|`0`||
 |`--p2p.ban.peers`|Enables peer banning. This should ONLY be enabled once certain peer scoring is working correctly.|||
 |`--p2p.bootnodes`|Comma-separated base64-format ENR list. Bootnodes to start discovering other node records from.|||
-|`--p2p.disable`|Completely disable the P2P stack|||
+|`--p2p.disable`|Completely disable the P2P stack, if p2p is disabled, mining will also be disabled.|||
```
Comment (qzhodl) [inline README.md:94]: I think the more accurate expression maybe "If P2P is disabled, mining will not start automatically"?

**Pull request merged**